File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.py
from collections import defaultdict
def solution(participant, completion):
    answer = ''
    
    compDict = defaultdict(int)
    for c in completion:
        compDict[c] += 1
    
    for p in participant:
        compDict[p] -= 1
        if compDict[p] < 0:
            answer = p
            break
    
    
    return answer

File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.java
import java.util.*;

class Solution {
    public String solution(String[] participant, String[] completion) {
        String answer = "";
        
        Map<String,Integer> completionMap = new HashMap<>();
        
        for(String c : completion){
            completionMap.put(c, completionMap.getOrDefault(c,0) + 1);
        }
        
        for(String p : participant){
            if(completionMap.getOrDefault(p,0) <= 0){
                answer = p;
                break;
            }
            
            completionMap.put(p, completionMap.get(p) - 1);
        }
        
        return answer;
    }
}